src/core/gltf2importer/gltf2parser.cpp

```diff
@@ -149,8 +149,38 @@ Qt3DCore::QEntity *GLTF2Parser::parse(const QString &filePath)
     }
 
     QFileInfo finfo(filePath);
-    const QByteArray jsonData = f.readAll();
-    return parse(jsonData, finfo.absolutePath(), finfo.fileName());
+    bool isBinary = true;
+    struct GLBHeader {
+        quint32 magic;
+        quint32 version;
+        quint32 length;
+    } glbHeader;
+
+    if (f.read(reinterpret_cast<char *>(&glbHeader), sizeof(glbHeader)) != sizeof(glbHeader)) {
+        isBinary = false;
+    } else {
+        if (glbHeader.magic != 0x46546C67) {
```

Comment:
Would it be cleaner to move the glbHeader check into a dedicated function (that can be more easily tested in a standalone way) ?

Reply:
Using a const variable for the magic like we do for the JSON keys would also be nicer

Reply:
done